addons/l10n_account_withholding_tax/tests/test_account_withholding_flows.py

```diff
@@ -1068,3 +1068,94 @@ def test_partner_is_set_on_lines(self):
+            "name": "+Base Tag",
+            "applicability": "taxes",
+        })
+        base_tag_minus = self.env["account.account.tag"].create({
+            "name": "-Base Tag",
+            "applicability": "taxes",
+            "tax_negate": True,
+        })
+        tax_tag_plus = self.env["account.account.tag"].create({
+            "name": "+Tax Tag",
+            "applicability": "taxes",
+        })
+        tax_tag_minus = self.env["account.account.tag"].create({
+            "name": "-Tax Tag",
+            "applicability": "taxes",
+            "tax_negate": True,
+        })
+        withholding_tax = self.percent_tax(
+            amount=-1,
+            type_tax_use='purchase',
+            is_withholding_tax_on_payment=True,
+            withholding_sequence_id=self.withholding_sequence.id,
+            invoice_repartition_line_ids=[
+                Command.create({'repartition_type': 'base', 'factor_percent': 100.0, 'tag_ids': base_tag_plus.ids}),
+                Command.create({'repartition_type': 'tax', 'factor_percent': 100.0, 'tag_ids': tax_tag_plus.ids}),
+            ],
+            refund_repartition_line_ids=[
+                Command.create({'repartition_type': 'base', 'factor_percent': 100.0, 'tag_ids': base_tag_minus.ids}),
+                Command.create({'repartition_type': 'tax', 'factor_percent': 100.0, 'tag_ids': tax_tag_minus.ids}),
+            ],
+        )
+        invoice = self.env["account.move"].create({
+            "move_type": "in_invoice",
+            "partner_id": self.partner_a.id,
+            "invoice_date": "2025-01-01",
+            "invoice_line_ids": [
+                Command.create(
+                    {
+                        "name": 'WTH Line',
+                        "price_unit": 1000.0,
+                        "tax_ids": [Command.set(withholding_tax.ids)],
+                    }
+                )
+            ],
+        })
+        invoice.action_post()
+
+        payment_register = (
+            self.env["account.payment.register"]
+            .with_context(active_model="account.move", active_ids=invoice.ids)
+            .create({})
+        )
+        payment = payment_register._create_payments()
+        self.assertRecordValues(payment.move_id.line_ids, [
+            # Liquidity line:
+            {'balance': -990.0,     'tax_tag_ids': []},
+            # Receivable line:
+            {'balance': 1000.0,    'tax_tag_ids': []},
+            # withholding line:
+            {'balance': -10.0,       'tax_tag_ids': tax_tag_plus.ids},
+            # base lines:
+            {'balance': -1000.0,     'tax_tag_ids': base_tag_plus.ids},
+            {'balance': 1000.0,    'tax_tag_ids': []},
+        ])
+
+        move_reversal = self.env['account.move.reversal'].with_context(active_model="account.move", active_ids=invoice.ids).create({
+            'date': fields.Date.from_string('2025-02-01'),
+            'journal_id': invoice.journal_id.id,
+        })
```

Comment:
Would be nice to be consistent with indentation used for `payment_register` (same for the use of single/double quotes) :nerd_face: